Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -433,6 +433,18 @@ void FrameLoader::checkContentPolicy(const ResourceResponse& response, PolicyChe
     client().dispatchDecidePolicyForResponse(response, activeDocumentLoader()->request(), identifier, activeDocumentLoader()->downloadAttribute(), WTFMove(function));
 }
 
+bool FrameLoader::upgradeRequestforHTTPSOnlyIfNeeded(const URL& originalURL, URL& newURL) const
+{
+    if (m_documentLoader && m_documentLoader->networkConnectionIntegrityPolicy().contains(NetworkConnectionIntegrity::HTTPSOnly) && newURL.protocolIs("http"_s) && ((RegistrableDomain(newURL) == RegistrableDomain(originalURL) && !m_documentLoader->networkConnectionIntegrityPolicy().contains(NetworkConnectionIntegrity::HTTPSOnlyExplicitlyBypassedForDomain)) || RegistrableDomain(newURL) != RegistrableDomain(originalURL))) {
+        FRAMELOADER_RELEASE_LOG(ResourceLoading, "upgradeRequestforHTTPSOnlyIfNeeded: upgrading navigation request");
+        newURL.setProtocol("https"_s);
+        if (newURL.port() && newURL.port().value() == 80)
+            newURL.setPort(443);
```

Comment:
if (newURL.port() == 80)
    newURL.setPort(std::nullopt);

Same in ContentSecurityPolicy::upgradeInsecureRequestIfNeeded